internal/controller/storageclient_controller.go

```diff
@@ -779,27 +661,31 @@ func removeStorageClaimAsOwner(obj client.Object) {
 	}
 }
 
-func (r *storageClientReconcile) EnsureClusterScopedResource(obj client.Object, extRes *providerpb.ExternalResource, useReplace bool) error {
-	obj.SetName(extRes.Name)
+func (r *storageClientReconcile) reconcileResource(obj client.Object, rawObject []byte, useReplace bool) error {
+	objectMeta := metav1.ObjectMeta{}
+	if err := json.Unmarshal(rawObject, &objectMeta); err != nil {
+		return err
+	}
+
+	obj.SetName(objectMeta.Name)
+	obj.SetNamespace(objectMeta.Namespace)
 
 	mutateFunc := func() error {
-		if err := json.Unmarshal(extRes.Data, obj); err != nil {
+		if err := json.Unmarshal(rawObject, obj); err != nil {
```

Comment:
I believe we agreed to a comment here explaining the merge nature of unmarshal which is necessary for the correctness of this code